coordinator/txmanager.go

```diff
@@ -323,16 +329,21 @@ func (t *TxManager) checkEthTransactionReceipt(ctx context.Context, batchInfo *B
 func (t *TxManager) handleReceipt(ctx context.Context, batchInfo *BatchInfo) (*int64, error) {
 	receipt := batchInfo.Receipt
 	if receipt != nil {
-		if batchInfo.EthTx.Nonce()+1 > t.accNonce {
-			t.accNonce = batchInfo.EthTx.Nonce() + 1
+		// taking last eth tx nonce as it probably have the highest nonce
+		lastEthTx := batchInfo.EthTxs[len(batchInfo.EthTxs)-1]
```

Comment:
Is it okay, that I take lastEthTx to get nonce?

Reply:
Yes it is, because in this task we will not change the way hermeznode manage the nonce.
So we just need to make sure this change does not break how hermeznode manage the nonces and tx receipts nowadays.
This change seems to be correct for me but @ed255 's second opinion is very valid too.